standalone_self_upgrade.cf.in

```diff
@@ -115,11 +115,38 @@ bundle agent cfengine_software
 # @brief Variables to control the specifics in desired package selection
 {
   vars:
+    any::
+
+      # Extract the hub binary version info if it's available. Only expected to
+      # be available on a client.
+      "hub_binary_version" -> { "ENT-10664" }
+        data => data_regextract(
+                                 "^(?<major_minor_patch>\d+\.\d+\.\d+)-(?<release>\d+)",
+                                 readfile("$(sys.statedir)$(const.dirsep)hub-cf_version.txt" ) ),
+        if => fileexists( "$(sys.statedir)$(const.dirsep)hub-cf_version.txt" );
 
       # Default desired CFEngine software
       "pkg_name" string => ifelse( isvariable( "def.cfengine_software_pkg_name" ), $(def.cfengine_software_pkg_name), "cfengine-nova");
-      "pkg_version" string => ifelse( isvariable( "def.cfengine_software_pkg_version" ), $(def.cfengine_software_pkg_version), "@VERSION@");
-      "pkg_release" string => ifelse( isvariable( "def.cfengine_software_pkg_release" ), $(def.cfengine_software_pkg_release), "@RELEASE@");
+
+
+      "pkg_version" -> { "ENT-10664" }
+        string => "$(sys.cf_version_major).$(sys.cf_version_minor).$(sys.cf_version_patch)",
+        if => "am_policy_hub|policy_server",
+        comment => "The hub will use it's own version to seed client packages.";
+      "pkg_version" -> { "ENT-10664" }
+        string => "$(hub_binary_version[major_minor_patch])",
+        if => isvariable("hub_binary_version[major_minor_patch]"),
```

Comment:
here and below, won't `hub_binary_version` exist on the hub as well so will use this pkg_version instead of the one above where you mention `if => "am_policy_hub|policy_server"`?

Reply:
I see, hub-cf_version.txt is only copied for non-hubs. 👍

Reply:
Right, might be more clear if cf_version.txt was called my-cf_version.txt perhaps.